bin/export-db-to-sqlite.sh

```diff
@@ -7,125 +7,305 @@
+
+# Convert the array into a comma-separated string suitable for the SQL IN clause
+tables_list=$(printf "'%s'," "${tables_to_wipe_after_import[@]}")
+tables_list=${tables_list%,}  # Remove the trailing comma
+
+# Execute the SQLite commands
+sqlite3 "$output_db" <<EOF
+BEGIN;
+
+-- Step 1: Create a temporary table to store the relations
+CREATE TEMPORARY TABLE fk_relations AS
+SELECT
+    fk_table.name AS table_name,
+    fk."from" AS fk_column
+FROM
+    sqlite_master AS fk_table
+JOIN
+    pragma_foreign_key_list(fk_table.name) AS fk
+WHERE
+    fk."table" IN ($tables_list)
+AND
+    fk_table.type = 'table';
+
+-- Step 2: Generate and execute UPDATE statements in a loop
+-- Output the UPDATE statements and execute them immediately
+.output $nullify_specific_columns_sql
+SELECT
+    'UPDATE "' || table_name || '" SET "' || fk_column || '" = NULL;'
+FROM
+    fk_relations;
+.output stdout
+
+-- Step 3: Execute the generated SQL statements
+.read $nullify_specific_columns_sql
+
+COMMIT;
+EOF
+
+echo "This is the SQL we ran to null out the columns:"
+cat $nullify_specific_columns_sql || all_well=false
+check_status_and_handle_failure "Showing temporary SQL file"
+
+rm -f $nullify_specific_columns_sql || all_well=false
+check_status_and_handle_failure "Removing temporary SQL file"
+echo "Deleted that temporary sql"
+
+# Delete rows from tables mentioned in tables_to_wipe_after_import
+for table in "${tables_to_wipe_after_import[@]}"
+do
+    sqlite3 $output_db "DELETE FROM $table"
+    echo "Purged now-redundant data from: $table"
+done
 
-PROD_DETAILS_STORAGE=product_details.storage.PDFileStorage \
-    DATABASE_URL=sqlite://$output_db \
-    python manage.py loaddata "/tmp/export_remainder.json" || all_well=false
+python manage.py rebuild_references_index
+check_status_and_handle_failure "Running rebuild_references_index"
+
+python manage.py wagtail_update_index
+check_status_and_handle_failure "Running wagtail_update_index"
+echo "Rebuilt Wagtail object reference helper and search index"
+
+# Check if tables in tables_to_wipe_after_import are empty
+for table in "${tables_to_wipe_after_import[@]}"
+do
+    count=$(sqlite3 $output_db "SELECT COUNT(*) FROM $table")
+    if [[ $count -ne 0 ]]; then
+        echo "ERROR: Table $table is not empty"
+        all_well=false
+    fi
+done
+echo "Checked that the tables we expect to be empty are empty"
 
-check_status 96
+check_status_and_handle_failure "Seeking expected empty tables"
 
-# 5. There are things we can't omit or redact in the steps above, so
-# load the DB and programatically prune it, too
-DATABASE_URL=sqlite://$output_db python manage.py scrub_exported_cms_data || all_well=false
+export DATABASE_URL=$ORIGINAL_DATABASE_URL
+echo "Restored original DATABASE_URL to $DATABASE_URL"
 
-check_status 95
+check_status_and_handle_failure "Checking all_well at the end of the run"
 
 echo "Export to $output_db successful"
```

Comment:
So much going on in here but I read it top to bottom and it seemed to make sense. Lots of data munging. A little surprised at how much detail there is to deal with (e.g. nullifying foreign keys) but kudos for digging into those details.

Reply:
Yeah, if `dumpdata` took the `--exclude` option and not only omitted tables but also nullified/sipped the FKs that referenced those excluded tables, this would have been basically all Django management command calls

Reply:
is it because the foreign keys might include PII?

Reply:
The tables the FKs linked to would have contained PII for `auth.User` and potentially sensitive data (eg draft pages that hadn't been legally reviewed yet) for `wagtail.Revision`. 

I had two reasons for nullifying - maybe neither is particularly high risk, but I also want to avoid Least Surprise:

1) There is implicit data leakage in seeing which pages were owned by user with an id of X -- someone could infer from that how many people have CMS access, who (pseudonymously) controls which areas, etc.

2) If we don't nullify them here, the records could end up linking to unexpected users/pages/other tables once  someone using the export starts to add their own data in local dev, which could cause confusion and waste time. (e.g. without the nullifying, `wagtailcore_page` will still have the `owner_id` set to, say, `2`, which may or may not exist, but if it eventually does exist locally (eg someone makes a superuser (`1`) and a regular user (`2`), it won't be the same user that originally had a pk of 2.). On reflection, I don't VACUUM after deleting the users, so maybe this would never happen as the autoincrement won't get reset...

Reply:
makes sense